src/utils/native.ts

```diff
@@ -158,31 +158,57 @@ export function schnorrkelVerify(
 	return EthkeyBridge.schnorrkelVerify(seed, message, signature);
 }
 
-// Decrypt data and return a reference to the result
-export function decryptDataRef(
-	data: string,
-	password: string
-): Promise<number> {
-	return EthkeyBridge.decryptDataRef(data, password);
-}
+export class SeedRef {
+	private dataRef: number;
+	private valid: boolean;
 
-// Clean up the memory allocation made by decryptDataRef
-export function destroyDataRef(dataRef: number): Promise<number> {
-	return EthkeyBridge.destroyDataRef(dataRef);
-}
+	constructor() {
+		this.dataRef = 0;
+		this.valid = false;
+	}
 
-// Use a reference returned by decryptDataRef to sign a message
-export function brainWalletSignWithRef(
-	seedRef: number,
-	message: string
-): Promise<string> {
-	return EthkeyBridge.brainWalletSignWithRef(seedRef, message);
-}
+	// Decrypt a seed and store the reference. Must be called before signing.
+	tryCreate(encryptedSeed: string, password: string): Promise<void> {
+		if (this.valid) {
```

Comment:
```suggestion
		if (!this.valid) {
```

Reply:
The line is correct as it is.  `tryCreate` should *fail* if the reference *is* valid.  If the reference is valid and we create another one, the old one will leak.

On the other hand, `tryDestroy` and signing functions should *fail* if the reference is *not* valid.

Reply:
yeah, my mistake